### PR TITLE
Simpler PML for H field

### PR DESCRIPTION
**Pull request by ajnonaka**
This PR implements PML boundary conditions for the H field for simulations with USE_LLG_TRUE.
The goal is to use the PML data structures and kernels for the B field PML treatment.
The main things to remember are that when doing the E PML update that H needs to be passed in and mu0 scaling appropriately applied, and when doing the B PML update a mu0 scaling needs to be appropriately applied.

### CODE REVIEW

File: Source/FieldSolver/WarpXPushFieldsEM.cpp [hunk 1]
```diff
@@ -446,8 +446,16 @@ WarpX::MacroscopicEvolveHM (int lev, PatchType patch_type, amrex::Real a_dt) {
     else {
         amrex::Abort("Macroscopic EvolveHM is not implemented for lev > 0 yet");
     }
-    if (do_pml) {
-        amrex::Abort("Macroscopic EvolveHM is not implemented for pml boundary condition yet");
+
+    // Evolve H field in PML cells
```
Comment (ajnonaka) [inline Source/FieldSolver/WarpXPushFieldsEM.cpp:380]:
```suggestion
    // Evolve H field in PML cells
    // We use the data structures and kernels for the B PML update for simplicity.
```

File: Source/FieldSolver/WarpXPushFieldsEM.cpp [hunk 2]
```diff
@@ -479,8 +487,16 @@ WarpX::MacroscopicEvolveHM_2nd (int lev, PatchType patch_type, amrex::Real a_dt)
     else {
         amrex::Abort("Macroscopic EvolveHM_2nd is not implemented for lev > 0 yet");
     }
-    if (do_pml) {
-        amrex::Abort("Macroscopic EvolveHM_2nd is not implemented for pml boundary condition yet");
+
+    // Evolve H field in PML cells
```
Comment (ajnonaka) [inline Source/FieldSolver/WarpXPushFieldsEM.cpp:422]:
```suggestion
    // Evolve H field in PML cells
    // We use the data structures and kernels for the B PML update for simplicity.
```

File: Source/Parallelization/WarpXComm.cpp [hunk 3]
```diff
@@ -596,15 +581,12 @@ WarpX::FillBoundaryH (int lev, PatchType patch_type, IntVect ng)
     {
         if (do_pml && pml[lev]->ok())
         {
-            amrex:Abort("PML not included for EvolveHM yet");
-            /*
-            pml[lev]->ExchangeH(patch_type,
+            pml[lev]->ExchangeB(patch_type,
```
Comment (ajnonaka) [inline Source/Parallelization/WarpXComm.cpp:588]:
```suggestion
            // We use the data structures and kernels for the B PML update for simplicity.
            // But here we exchange with the Hfield
            pml[lev]->ExchangeB(patch_type,
```

File: Source/Parallelization/WarpXComm.cpp [hunk 4]
```diff
@@ -622,15 +604,14 @@ WarpX::FillBoundaryH (int lev, PatchType patch_type, IntVect ng)
     else if (patch_type == PatchType::coarse)
     {
         amrex::Abort("EvolveHM does not come with coarse patch yet");
-        /*
         if (do_pml && pml[lev]->ok())
         {
-        pml[lev]->ExchangeH(patch_type,
+        pml[lev]->ExchangeB(patch_type,
```
Comment (ajnonaka) [inline Source/Parallelization/WarpXComm.cpp:615]:
```suggestion
        // We use the data structures and kernels for the B PML update for simplicity.
        // But here we exchange with the Hfield
        pml[lev]->ExchangeB(patch_type,
```

File: Source/FieldSolver/FiniteDifferenceSolver/EvolveBPML.cpp [hunk 5]
```diff
@@ -61,6 +61,12 @@ void FiniteDifferenceSolver::EvolveBPMLCartesian (
     std::array< amrex::MultiFab*, 3 > const Efield,
     amrex::Real const dt ) {
 
+#if WARPX_MAG_LLG
+    Real factor = 1./PhysConst::mu0;
+#else
+    Real factor = 1.;
```
Comment (jackieyao0114) [inline Source/FieldSolver/FiniteDifferenceSolver/EvolveBPML.cpp:67]:
```suggestion
    Real factor = 1._rt;
```

Comment (jackieyao0114) [reply Source/FieldSolver/FiniteDifferenceSolver/EvolveBPML.cpp:67]: with the spirit of WarpX, added `_rt`

**Pull request closed without merging**